### ISSUE TITLE
Analyzer to detect unused CollectionDefinitions

### ISSUE THREAD

**Issue by marcind** [xunit/xunit#1277]
If there is a class marked with`CollectionDefinition` but no test class referencing it, flag it as a warning.
This can help avoid typos in the collection name.

**Comment by jzabroski** [comment]
As well as the corollary: if a Collection has a Name without an explicit CollectionDefinition, flag it as a warning.  Fixer: Apply a CollectionDefinition to the class if the Collection exists on only one class.  Fixer: Generate an empty class if the Collection exists on multiple classes.

**Comment by bradwilson** [comment]
I disagree with @jzabroski's desire to flag this as a warning. Offering a refactoring to generate a collection definition would be fine, but it's not a problem (therefore not worth of a Warning-level analyzer) to have collections without definitions.

**Comment by jzabroski** [comment]
Thanks for explaining that.  I can be overly tidy at times.  Agree, warning not needed, but refactoring suggestion to generate a collection definition would be nice.  That means it would need to flag as an Information-level analyzer, right?

**Comment by bradwilson** [comment]
Yeah, I assume an info-level analyzer is necessary to be able to offer the fixer as a refactoring operation.
